service/matching/matchingEngine_test.go

```diff
@@ -545,265 +511,207 @@ func (s *matchingEngineSuite) TestAddThenConsumeActivities() {
-						HeartbeatTimeoutSeconds:       common.Int32Ptr(1),
-					}),
-			}, nil
-		}).AnyTimes()
+	s.setupRecordTaskStartedMock(taskType, testParam, false)
 
 	pollFunc := func(maxDispatch float64) (*pollTaskResponse, error) {
 		pollReq := &pollTaskRequest{
 			TaskType:         taskType,
-			DomainUUID:       domainID,
-			TaskList:         taskList,
-			Identity:         identity,
+			DomainUUID:       testParam.DomainID,
+			TaskList:         testParam.TaskList,
+			Identity:         testParam.Identity,
 			TaskListMetadata: &types.TaskListMetadata{MaxTasksPerSecond: &maxDispatch},
 		}
 		return pollTask(s.matchingEngine, s.handlerContext, pollReq)
 	}
 
 	for i := int64(0); i < taskCount; i++ {
 		scheduleID := i * 3
-
 		var wg sync.WaitGroup
 		var result *pollTaskResponse
 		var pollErr error
-		maxDispatch := _defaultTaskDispatchRPS
-		if i == taskCount/2 {
-			maxDispatch = 0
-		}
 		wg.Add(1)
 		go func() {
 			defer wg.Done()
-			result, pollErr = pollFunc(maxDispatch)
+			result, pollErr = pollFunc(_defaultTaskDispatchRPS)
 		}()
-		time.Sleep(20 * time.Millisecond) // Necessary for sync match to happen
+		time.Sleep(20 * time.Millisecond) // Wait for a short period of time to let the poller start so that sync match will happen
 		addRequest := &addTaskRequest{
 			TaskType:                      taskType,
-			DomainUUID:                    domainID,
-			Execution:                     &workflowExecution,
+			DomainUUID:                    testParam.DomainID,
+			Execution:                     testParam.WorkflowExecution,
 			ScheduleID:                    scheduleID,
-			TaskList:                      taskList,
+			TaskList:                      testParam.TaskList,
 			ScheduleToStartTimeoutSeconds: 1,
 		}
 		_, err := addTask(s.matchingEngine, s.handlerContext, addRequest)
 		wg.Wait()
 		s.NoError(err)
 		s.NoError(pollErr)
 		s.NotNil(result)
+		s.assertPollTaskResponse(taskType, testParam, scheduleID, result)
+	}
 
-		if isEmptyToken(result.TaskToken) {
-			// when ratelimit is set to zero, poller is expected to return empty result
-			// reset ratelimit, poll again and make sure task is returned this time
-			s.logger.Debug("empty poll returned")
-			s.Equal(float64(0), maxDispatch)
-			maxDispatch = _defaultTaskDispatchRPS
-			wg.Add(1)
-			go func() {
-				defer wg.Done()
-				result, pollErr = pollFunc(maxDispatch)
-			}()
-			wg.Wait()
-			s.NoError(err)
-			s.NoError(pollErr)
-			s.NotNil(result)
-			s.False(isEmptyToken(result.TaskToken))
-		}
-
-		s.EqualValues(activityID, result.ActivityID)
-		s.EqualValues(activityType, result.ActivityType)
-		s.EqualValues(activityInput, result.Input)
-		s.EqualValues(workflowExecution, *result.WorkflowExecution)
-		token := &common.TaskToken{
-			DomainID:     domainID,
-			WorkflowID:   workflowID,
-			RunID:        runID,
-			ScheduleID:   scheduleID,
-			ActivityID:   activityID,
-			ActivityType: activityTypeName,
+	for i := int64(0); i < throttledTaskCount; i++ {
```

Comment:
It's not imediately clear what's going on here without being pretty steeped in the polling code. Do you mind either mind putting a comment or wrapping this in some function which indicates the intent. It's not immediately clear to me what throttledTaskCount is intended to be used for?